public/app/features/library-panels/components/DeleteLibraryPanelModal/DeleteLibraryPanelModal.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { FC, useEffect, useMemo, useReducer } from 'react';
+import { Button, HorizontalGroup, Modal, useStyles } from '@grafana/ui';
+import { LoadingState } from '@grafana/data';
+
+import { LibraryPanelDTO } from '../../types';
+import { asyncDispatcher } from '../LibraryPanelsView/actions';
+import { deleteLibraryPanelModalReducer, initialDeleteLibraryPanelModalState } from './reducer';
+import { getConnectedDashboards } from './actions';
+import { getModalStyles } from '../../styles';
+
+interface Props {
+  libraryPanel: LibraryPanelDTO;
+  onConfirm: () => void;
+  onDismiss: () => void;
+}
+
+export const DeleteLibraryPanelModal: FC<Props> = ({ libraryPanel, onDismiss, onConfirm }) => {
+  const styles = useStyles(getModalStyles);
+  const [{ dashboardTitles, loadingState }, dispatch] = useReducer(
+    deleteLibraryPanelModalReducer,
+    initialDeleteLibraryPanelModalState
+  );
+  const asyncDispatch = useMemo(() => asyncDispatcher(dispatch), [dispatch]);
+  useEffect(() => {
+    asyncDispatch(getConnectedDashboards(libraryPanel));
+  }, []);
+  const connected = Boolean(dashboardTitles.length);
+  const done = loadingState === LoadingState.Done;
+
+  return (
+    <Modal className={styles.modal} title="Delete library panel" icon="trash-alt" onDismiss={onDismiss} isOpen={true}>
```

Comment:
what is these styles.modal & getModalStyles? does the modal not have this built-in?

Reply:
styles.modal is just for the width, looked super wide without it.

Reply:
Ah, I see now itä's only for the width. the getModalStyles made me think it was some global modal styles that had to be passed in.

Reply:
Yepp, it shared between 2 LibaryPanel modals.